runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -7150,7 +7157,7 @@ void local_yield(void)
     //
     PERFSTATS_INC(tskyield_in_lyield_no_cd_cnt);
     if (can_task_yield()) 
-      chpl_task_yield();
+      chpl_task_yield2(cnt);
     else
       sched_yield();
```

Comment:
Is there no need for QSBR checkpointing if a pthread not hosting a task yields, as here?  Note that the pthread may have run Chapel code in the past.  I don't understand privatization's use of QSBR well enough to know if that's an issue.

Reply:
If a pthread has registered itself, you're right it should invoke a checkpoint.

Reply:
Okay, in this case where we're talking about the AM handler pthread in the comm layer, it can execute Chapel code from time to time (not whole tasks, but small on-stmt bodies for "fast" executeOn operations).  So it will need to register itself.  This would be in `polling_task()`, probably after the `set_up_for_polling()` call.

Reply:
In `polling_task` is there a reason it does not invoke `local_yield`?

Reply:
It just saves a teensy amount of yield latency and thus AM latency.  We know that the polling task has a comm domain (`cd != NULL`), so if it did call `local_yield()`, that would in turn call `sched_yield()`.  That being so, the polling task might as well call `sched_yield()` itself.